zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/RealmAwareZkClientTestBase.java

```diff
@@ -67,4 +76,132 @@ public void afterClass() {
+
+    //Create Nodes
+    List<Op> ops = Arrays.asList(
+            Op.create(PARENT_PATH, new byte[0],
+                    ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT),
+            Op.create(PARENT_PATH + test_name, new byte[0],
+                    ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT));
+
+    //Execute transactional support on operations and verify they were run
+    List<OpResult> opResults = _realmAwareZkClient.multi(ops);
+    Assert.assertTrue(opResults.get(0) instanceof OpResult.CreateResult);
+    Assert.assertTrue(opResults.get(1) instanceof OpResult.CreateResult);
+
+    //Verify that the znodes were created
+    Assert.assertTrue(_realmAwareZkClient.exists(PARENT_PATH), "Path has not been created.");
+    Assert.assertTrue(_realmAwareZkClient.exists(PARENT_PATH + test_name), "Path has not been created.");
+
+    cleanup();
+  }
+
+  /**
+   * Multi should be an all or nothing transaction. Creating correct
+   * paths and a singular bad one should all fail.
+   */
+  @Test(dependsOnMethods = "testMultiCreate")
+  public void testMultiFail() {
+    String test_name = "/test_multi_fail";
+    //Create Nodes
+    List<Op> ops = Arrays.asList(
+            Op.create(PARENT_PATH, new byte[0],
+                    ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT),
+            Op.create(PARENT_PATH + test_name, new byte[0],
+                    ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT),
+            Op.create(TEST_INVALID_PATH, new byte[0],
+                    ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT));
+    try {
+      _realmAwareZkClient.multi(ops);
+      Assert.fail("Should have thrown an exception. Cannot run multi on incorrect path.");
+    } catch (Exception e) {
+      boolean pathExists = _realmAwareZkClient.exists(PARENT_PATH);
+      Assert.assertFalse(pathExists, "Path should not have been created.");
+
+      cleanup();
+    }
+  }
+
+  /**
+   * Test that zk multi works for delete.
+   */
+  @Test(dependsOnMethods = "testMultiFail")
+  public void testMultiDelete() {
+    String test_name = "/test_multi_delete";
+    //Create Nodes
+    List<Op> ops = Arrays.asList(
+            Op.create(PARENT_PATH, new byte[0],
+                    ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT),
+            Op.create(PARENT_PATH + test_name, new byte[0],
+                    ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT),
+            Op.delete(PARENT_PATH + test_name, -1));
+
+    List<OpResult> opResults = _realmAwareZkClient.multi(ops);
+    Assert.assertTrue(opResults.get(0) instanceof OpResult.CreateResult);
+    Assert.assertTrue(opResults.get(1) instanceof OpResult.CreateResult);
+    Assert.assertTrue(opResults.get(2) instanceof OpResult.DeleteResult);
+
+    Assert.assertTrue(_realmAwareZkClient.exists(PARENT_PATH), "Path has not been created.");
+    Assert.assertFalse(_realmAwareZkClient.exists(PARENT_PATH + test_name), "Path should have been removed.");
```

Comment:
I guess the message should be the opposite?

Reply:
I think the message only shows when the condition isn't true. For example, if PARENT_PATH does not exist, then it would error with "Path has not been created"